libraries/AP_Math/vector2.cpp

```diff
@@ -237,8 +248,175 @@ bool Vector2<T>::circle_segment_intersection(const Vector2<T>& seg_start, const
     return false;
 }
 
+// returns the normalized vector
+template <typename T>
+Vector2<T> Vector2<T>::normalized() const
+{
+    return *this/length();
+}
+
+template <typename T>
+void Vector2<T>::normalize()
+{
+    *this /= length();
+}
+
+// reflects this vector about n
+template <typename T>
+void Vector2<T>::reflect(const Vector2<T> &n)
+{
+    Vector2<T> orig(*this);
+    project(n);
+    *this = *this*2 - orig;
+}
+
+// projects this vector onto v
+template <typename T>
+void Vector2<T>::project(const Vector2<T> &v)
+{
+    *this= v * (*this * v)/(v*v);
+}
+
+// returns this vector projected onto v
+template <typename T>
+Vector2<T> Vector2<T>::projected(const Vector2<T> &v)
+{
+    return v * (*this * v)/(v*v);
+}
+
+// given a position p1 and a velocity v1 produce a vector
```

Comment:
Description doesn't match the function arguments. Leaves you guessing that `p1` is `pos_delta`

Reply:
I'm pretty sure this was an existing issue with the comments before the function was moved but I'll fix it anyway, txs.

Reply:
Quite possibly, I didn't catch until later that a lot of stuff in this file was moving it out of the header or not, which meant I had started reviewing old/pre existing code/made it harder to see what was actually a change.

Reply:
sorry about that.. I took a short-cut when merging from the OBC branch because I just took the complete file instead of trying to find all the patches that made up the change.

Reply:
Ah. That's really bad actually. Means this needs a much more detailed review to catch things that actually changed between master and this file (IE the problem introduced back into the polygon code)

Reply:
ok, I'll go back and split the changes into two commits - one for just the code moving, another for he actual change.